src/unix/pipe.c

```diff
@@ -287,6 +287,62 @@ int uv_pipe_getpeername(const uv_pipe_t* handle, char* buffer, size_t* size) {
 }
 
 
+int uv_pipe_get_credentials(const uv_pipe_t* handle, uv_credentials_t* creds) {
+  int r;
+  uv_os_fd_t os_fd;
+
+  r = uv_fileno((uv_handle_t*) handle, &os_fd);
+  if (r < 0)
+    return r;
+
+#if defined(__APPLE__)
```

Comment:
Should FreeBSD be using this path also? How about other non-Linux platforms?

Reply:
Oops didn't mean to resolve, sorry.

Because uid_t and gid_t are unsigned, and on Windows I wanted to return -1 to match the behavior of the getpasswd function.

Reply:
Yes. Just installed FreeBSD and there's another problem: Nowhere in the FreeBSD API can you get the PID of a connected peer.

Should the PID be ifdef'd out and return `-1` with a documentation notice?

Reply:
Yes please.